kvbc/include/merkle_tree_db_adapter.h

```diff
@@ -182,9 +183,11 @@ class DBAdapter : public IDbAdapter {
 
   block::detail::Node getBlockNode(BlockId blockId) const;
 
-  KeysVector staleIndexKeysForVersion(const sparse_merkle::Version &version) const;
+  KeysVector staleIndexNonProvableKeysForBlock(BlockId blockId) const;
 
-  KeysVector internalKeysForVersion(const sparse_merkle::Version &version) const;
+  KeysVector staleIndexProvableKeysForVersion(const sparse_merkle::Version &version) const;
+
+  KeysVector internalProvableKeysForVersion(const sparse_merkle::Version &version) const;
```

Comment:
Nit: In terms of naming, I was thinking that maybe we can separate keys to:

- internalTreeKeysForVersion
- staleTreeKeysForVersion
- staleNonProvableKeysForBlock

Rationale is that internal keys are always tree ones and, technically, are not provable. Just something that came to mind.

Reply:
I am afraid this PR is being in review for a long time.
If you don't mind we can change it as a separate PR.